Tests/merge_test.py

```diff
@@ -1,17 +1,256 @@
+		self.assertIn(self.table2.table.ScriptList.ScriptRecord[0], table.ScriptList.ScriptRecord)
+
+		# Checks LookupList
+		self.assertEqual(table.LookupList.LookupCount, 2)
+		self.assertIn(self.table1.table.LookupList.Lookup[0], table.LookupList.Lookup)
+		self.assertIn(self.table2.table.LookupList.Lookup[0], table.LookupList.Lookup)
+
+		# Checks FeatureList
+		self.assertEqual(table.FeatureList.FeatureCount, 2)
+		self.assertIn(self.table1.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+		self.assertIn(self.table2.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+
+	def test_GSUB_merge_with_dups(self):
+		self.merger.duplicateGlyphsPerFont = [{}, {'uni0032#0': 'uni0032#1'}]
+		self.preMerge(self.table1)
+		self.preMerge(self.table2)
+
+		self.mergedTable.merge(self.merger, [self.table1, self.table2])
+		table = self.mergedTable.table
+
+		# Checks ScriptList
+		self.assertEqual(table.ScriptList.ScriptCount, 2)
+		self.assertIn(self.table1.table.ScriptList.ScriptRecord[0], table.ScriptList.ScriptRecord)
+		self.assertIn(self.table2.table.ScriptList.ScriptRecord[0], table.ScriptList.ScriptRecord)
+
+		# Checks LookupList
+		self.assertEqual(table.LookupList.LookupCount, 3)
+		self.assertIn(self.table1.table.LookupList.Lookup[0], table.LookupList.Lookup)
+		self.assertIn(self.table2.table.LookupList.Lookup[0], table.LookupList.Lookup)
+		self.assertEqual(table.LookupList.Lookup[-1].LookupFlag, 0)
+		self.assertEqual(table.LookupList.Lookup[-1].LookupType, 1)
+		self.assertEqual(table.LookupList.Lookup[-1].SubTableCount, 1)
+		self.assertEqual(table.LookupList.Lookup[-1].SubTable[0].mapping, self.merger.duplicateGlyphsPerFont[1])
+
+		# Checks FeatureList
+		self.assertEqual(table.FeatureList.FeatureCount, 3)
+		self.assertIn(self.table1.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+		self.assertIn(self.table2.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+		self.assertEqual(table.FeatureList.FeatureRecord[-1].FeatureTag, 'locl')
+
+	def test_GSUB_merge_with_same_dups_and_no_second_GSUB(self):
+		self.table2 = NotImplemented
+		self.merger.duplicateGlyphsPerFont = [{}, {'uni0032#0': 'uni0032#1'}]
+		self.merger.fonts = []
+		with mockIsGlyphSame(True):
+			self.mergedTable.merge(self.merger, [self.table1, self.table2])
+		table = self.mergedTable.table
+		self.assertEqual(table, self.table1.table)
+
+	def test_GSUB_merge_with_different_dups_and_no_second_GSUB(self):
+		self.table2 = NotImplemented
+		self.merger.duplicateGlyphsPerFont = [{}, {'uni0032#0': 'uni0032#1'}]
+		self.merger.fonts = []
+		self.merger.fontfiles = ['font1', 'font2']
+		with mockIsGlyphSame(False):
+			with CapturingLogHandler(fontTools.merge.log, "WARNING") as captor:
+				self.mergedTable.merge(self.merger, [self.table1, self.table2])
+
+		table = self.mergedTable.table
+		self.assertEqual(table, self.table1.table)
+		self.assertTrue(len([r for r in captor.records if r.msg != '']) > 0)
```

Comment:
@zjusbo this assert:

```python
self.assertTrue(len([r for r in captor.records if r.msg != '']) > 0)
```

would catch any non-empty (WARNING) messages; but I think you only want to capture those that match the expected substring, e.g.:

```python
self.assertTrue(any("is dropped and replaced by" in r.msg for r in captor.records))
```